pkg/monitoring/prometheusagent/sharding/sharding.go

```diff
@@ -40,7 +44,7 @@ func (s Strategy) ComputeShards(currentShardCount int, timeSeries float64) int {
 
 	// We always have a minimum of 1 agent, even if there is no worker node
 	if desiredShardCount <= 0 {
-		return 1
+		return math.Max(commonmonitoring.DefaultShards, currentShardCount)
```

Comment:
lol ... maybe the defaultshards should be defined here? Damn cycle

Reply:
Fixed it, but math.Max deals with float64, that's not pretty but should be fine

Reply:
Sure but I prefer that over yet another if 😅